include/deal.II/grid/filtered_iterator.h

```diff
@@ -753,6 +772,68 @@ make_filtered_iterator (const BaseIterator &i,
 
 
 
+#ifdef DEAL_II_WITH_CXX11
+/**
+ * Filter the  given range of iterators using a Predicate this allows to replace:
+ * @code
+ *   DoFHandler<dim> dof_handler;
+ *   ...
+ *   for (auto cell : dof_handler.active_cell_iterators())
+ *     {
+ *       if (cell->is_locally_owned())
+ *         {
+ *           fe_values.reinit (cell);
+ *           ...do the local integration on 'cell'...;
+ *         }
+ *     }
+ * @endcode
+ * by:
+ * @code
+ *   DoFHandler<dim> dof_handler;
+ *   ...
+ *   for (auto cell : filter_iterators(dof_handler.active_cell_iterators(),
+ *                                    IteratorFilters::LocallyOwned())
+ *     {
+ *       fe_values.reinit (cell);
+ *       ...do the local integration on 'cell'...;
+ *     }
+ * @endcode
+ *
+ * @author Bruno Turcksin @relates FilteredIterator @ingroup CPP11
+ */
+template <typename BaseIterator, typename Predicate>
+IteratorRange<FilteredIterator<BaseIterator> >
+filter_iterators (IteratorRange<BaseIterator> i,
+                  const Predicate &p)
+{
+  FilteredIterator<BaseIterator> fi(p, *(i.begin()));
+  FilteredIterator<BaseIterator> fi_end(p, *(i.end()));
+
+  return IteratorRange<FilteredIterator<BaseIterator> > (fi, fi_end);
+}
+#endif
+
+
+
+#ifdef DEAL_II_WITH_CXX14
```

Comment:
In C++11, the return type of a function cannot be ```auto``` but I don't know the return type until the function is called.